build.gradle

```diff
@@ -20,9 +20,8 @@ def includeResource(configuration) {
 def getVersionQualifier(String branch_name) {
     if(branch_name.startsWith('release'))
         return ''
-    if(branch_name == 'master')
-        return '-SNAPSHOT'
-    return '-' + branch_name + '-SNAPSHOT'
+    //Osgi in 5x accepts - in qualifier, 6x does not. Suggest removing branch section from snapshot qualifiers all together.
```

Comment:
Allowing feature branches to have the branchname in the artifact qualifier.

Reply:
To elaborate, since branch artifacts are not published, I think the risks for mixups with master artifacts is minimal. Seeing as the naming convention causes issues, I would just remove it instead of changing it.